include/fmt/format.h

```diff
@@ -1841,13 +1841,18 @@ OutputIt write(OutputIt out, const void* value) {
   return write_ptr<Char>(out, to_uintptr(value), nullptr);
 }
 
-template <typename Char, typename OutputIt, typename T>
-auto write(OutputIt out, const T& value) -> typename std::enable_if<
-    mapped_type_constant<T, basic_format_context<OutputIt, Char>>::value ==
-        type::custom_type,
-    OutputIt>::type {
-  basic_format_context<OutputIt, Char> ctx(out, {}, {});
-  return formatter<T>().format(value, ctx);
+template <typename Char, typename OutputIt, typename T,
+          typename Context = basic_format_context<OutputIt, Char>>
```

Comment:
I don't think context should be a template parameter.

Reply:
I did that to have one definition for `Context`, so it can be used in trailing return type too. But if it's a problem then I have 2 solution:

- by using `basic_format_context<OutputIt, Char>` at least in return type (or maybe in entire function)
- probably return type can be removed, but `enable_if` in it should be moved to template parameter and I'm not sure how old compilers will handle that

Reply:
I think it's better to have `basic_format_context<OutputIt, Char>` in two places as before.

Reply:
Ok, will do, but with conditional for `formatter_type` there will be more than 2 places

Reply:
I suggest adding an alias in a function. I just don't think it should be a template parameter.

Reply:
done